### PR TITLE
Replace neon node port

### PR DESCRIPTION
**Pull request by FloVanGH**
* Fixes all examples to works with node napi port
* Added libraryPaths setter to ComponentCompiler

### CODE REVIEW

File: api/node/src/timer.rs
Comment (ogoffart) [inline api/node/src/timer.rs:1]: this commit also add timers feature and test?

File: examples/carousel/ui/theme.slint [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // Copyright © SixtyFPS GmbH <info@slint.dev>
 // SPDX-License-Identifier: MIT
 
-import "fonts/Roboto-Regular.ttf";
-import "fonts/Roboto-Bold.ttf";
+import "../fonts/Roboto-Regular.ttf";
+import "../fonts/Roboto-Bold.ttf";
```
Comment (ogoffart) [inline examples/carousel/ui/theme.slint:5]: why this change?

Comment (FloVanGH) [reply examples/carousel/ui/theme.slint:5]: This change will be removed. I have trouble with relative file loading at the moment. With the fix this will be removed.

File: api/node/src/timer.rs
Comment (FloVanGH) [reply api/node/src/timer.rs:1]: Yes the timer was missing. We need it for the printerdemo. It was also a part of the old version.

File: api/node/package.json [hunk 2]
```diff
@@ -1,25 +1,49 @@
...
-        "typescript": "^4.0.3"
-    },
-    "scripts": {
-        "install": "neon build --release && tsc",
-        "build": "tsc",
-        "docs": "typedoc --hideGenerator --readme cover.md lib/index.ts"
-    },
-    "devDependencies": {
-        "typedoc": "^0.19.2"
+  "name": "slint-node",
```
[the remark this answers is not in the document]
Comment (FloVanGH) [reply api/node/package.json:2]: yes it's the name of the old `neon` version. This should be a replacement right?

File: api/node/README.md
[the remark this answers is not in the document]
Comment (FloVanGH) [reply api/node/README.md:1]: What do you mean?

[the remark this answers is not in the document]
Comment (FloVanGH) [reply api/node/README.md:1]: I see, sorry you mean the project README. You're right.

File: api/node/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (ogoffart) [reply api/node/package.json:2]: Previous name was "slint-ui"

File: api/node/README.md
[the remark this answers is not in the document]
Comment (FloVanGH) [reply api/node/README.md:1]: Done. The model example on the readme is already updated to the upcoming changes.

File: api/node/Cargo.toml [hunk 3]
```diff
@@ -27,6 +27,7 @@ slint-interpreter = { workspace = true, features = ["default", "display-diagnost
 spin_on = "0.1"
 css-color-parser2 = { workspace = true }
 itertools = { workspace = true }
+rand = "0.8.5"
```
Comment (ogoffart) [inline api/node/Cargo.toml:30]: I'm curious why this adds rand

File: api/node/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (FloVanGH) [reply api/node/package.json:2]: Ah right. That is in the package. Yes the name should be slint-ui

File: api/node/Cargo.toml (hunk 3 above)
Comment (FloVanGH) [reply api/node/Cargo.toml:30]: Thanks for the hint. That is not needed.

Comment (FloVanGH) [reply api/node/Cargo.toml]: removed

File: api/node/index.ts [hunk 4]
```diff
@@ -292,9 +294,10 @@ export class CompileError extends Error {
 /**
  * Loads the given slint file and returns a constructor to create an instance of the exported component.
  */
-export function loadFile(path: string) : Object {
+export function loadFile(filePath: string) : Object {
+    let absoluteFilePath = path.resolve(filePath);
     let compiler = new napi.ComponentCompiler;
-    let definition = compiler.buildFromPath(path);
+    let definition = compiler.buildFromPath(absoluteFilePath);
```
Comment (tronical) [inline api/node/index.ts:300]: I agree that this is a good workaround for now. Please add a comment here that explains why this is necessary. I'll look into fixing it.

File: api/node/index.ts [hunk 5]
```diff
@@ -305,7 +308,7 @@ export function loadFile(path: string) : Object {
         let errors = diagnostics.filter((d) => d.level == napi.DiagnosticLevel.Error);
 
         if (errors.length > 0) {
-            throw new CompileError("Could not compile " + path, errors);
+            throw new CompileError("Could not compile " + absoluteFilePath, errors);
```
Comment (tronical) [inline api/node/index.ts:311]: Given that making the path absolute is intended to be a workaround, I think here (and in the other place) it would be better to stick to using `path`.

File: api/node/index.ts [hunk 6]
```diff
@@ -356,6 +359,12 @@ export function loadFile(path: string) : Object {
     return slint_module;
 }
 
+export namespace Timer {
+    export function singleShot(duration: number, handler: () => void) {
+        napi.singleshotTimer(duration, handler)
+    }
+}
```
Comment (tronical) [inline api/node/index.ts:366]: Please add a comment here that this entire API should be removed. We should not need it, but we do until the event loop merge is ... merged.

File: api/node/package.json [hunk 7]
```diff
@@ -1,25 +1,49 @@
 {
-    "name": "slint-ui",
-    "version": "1.3.0",
-    "homepage": "https://github.com/slint-ui/slint",
```
Comment (tronical) [inline api/node/package.json:4]: Please re-add this field.

File: api/node/package.json [hunk 8]
```diff
@@ -1,25 +1,49 @@
...
-    "license": "SEE LICENSE IN LICENSE.md",
-    "repository": {
```
Comment (tronical) [inline api/node/package.json:6]: Please re-add this field.

File: api/node/package.json [hunk 9]
```diff
@@ -1,25 +1,49 @@
...
-        "build": "tsc",
-        "docs": "typedoc --hideGenerator --readme cover.md lib/index.ts"
-    },
-    "devDependencies": {
-        "typedoc": "^0.19.2"
+  "name": "slint-ui",
+  "version": "1.3.0",
+  "main": "index.js",
+  "types": "index.d.ts",
+  "license": "GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial",
```
Comment (tronical) [inline api/node/package.json:6]: Reading https://docs.npmjs.com/cli/v10/configuring-npm/package-json#license I think we need to use the old variant of this field. What do you think?

File: api/node/package.json [hunk 10]
```diff
@@ -1,25 +1,49 @@
...
+  "napi": {
+    "name": "slint-ui"
+  },
```
Comment (tronical) [inline api/node/package.json:9]: Do you know which part of napi-rs exactly needs this field? I wonder if this perhaps only needed for binary packages.

File: api/node/package.json [hunk 11]
```diff
@@ -1,25 +1,49 @@
...
+    "artifacts": "napi artifacts",
+    "compile": "esbuild index.ts --bundle --external:*.node --format=cjs --platform=node --outfile=index.js",
+    "build": "napi build --platform --release --js rust-module.js --dts rust-module.d.ts && npm run compile",
+    "build:debug": "napi build --platform --js rust-module.js --dts rust-module.d.ts && npm run compile && npm run syntax_check",
+    "install": "npm run build",
+    "docs": "npm run build && typedoc --hideGenerator --treatWarningsAsErrors --readme cover.md index.ts",
+    "prepublishOnly": "napi prepublish -t npm",
+    "test": "ava",
+    "syntax_check": "tsc -noEmit index.ts",
+    "universal": "napi universal",
```
Comment (tronical) [inline api/node/package.json:33]: Do you know if this is needed?

File: api/node/package.json [hunk 12]
```diff
@@ -1,25 +1,49 @@
...
+    "version": "1.3.0"
```
Comment (tronical) [inline api/node/package.json:34]: What is the purpose of this field in the "scripts" object?

File: api/node/src/interpreter/component_compiler.rs [hunk 13]
```diff
@@ -60,6 +61,16 @@ impl JsComponentCompiler {
             .collect()
     }
 
+    #[napi(setter)]
+    pub fn set_library_paths(&mut self, paths: HashMap<String, String>) {
```
Comment (tronical) [inline api/node/src/interpreter/component_compiler.rs:65]: IMO there should be a getter for this, as well as a test. This could be done separate from this PR. If yes, please file an issue.

File: tests/driver/nodejs/Cargo.toml [hunk 14]
```diff
@@ -14,12 +14,12 @@ publish = false
...
 [[bin]]
 path = "main.rs"
-name = "test-driver-nodejs"
+name = "test-driver-napi"
 
 [dev-dependencies]
 test_driver_lib = { path = "../driverlib" }
 # Require `artifact`  dependencies tracked by rust RFC 3028
-#slint-node = { path = "../../../api/node/native", artifact = ["cdylib"] }
+#slint-napi = { path = "../../../api/node", artifact = ["cdylib"] }
```
Comment (tronical) [inline tests/driver/nodejs/Cargo.toml:22]: Shouldn't that be `slint-node` again?

File: tests/driver/nodejs/Cargo.toml [hunk 15]
```diff
@@ -14,12 +14,12 @@ publish = false
 
 [[bin]]
 path = "main.rs"
-name = "test-driver-nodejs"
+name = "test-driver-napi"
```
Comment (tronical) [inline tests/driver/nodejs/Cargo.toml:17]: Further up in the diff you changed the CI instructions to run this command:

```
run: cargo test --verbose --all-features -p test-driver-nodejs -p slint-node
```

(used to be `-p test-driver-napi`)

Why do you rename the binary to not match the package name?

File: api/node/index.ts (hunk 4 above)
Comment (FloVanGH) [reply api/node/index.ts:302]: done

File: api/node/index.ts (hunk 5 above)
Comment (FloVanGH) [reply api/node/index.ts]: done

File: api/node/index.ts (hunk 6 above)
Comment (FloVanGH) [reply api/node/index.ts:370]: done

File: api/node/package.json (hunk 8 above)
Comment (FloVanGH) [reply api/node/package.json:6]: done-

File: api/node/package.json (hunk 7 above)
Comment (FloVanGH) [reply api/node/package.json:4]: done

File: api/node/src/interpreter/component_compiler.rs (hunk 13 above)
Comment (FloVanGH) [reply api/node/src/interpreter/component_compiler.rs:65]: It's an preparation to use it from js. But to finish it we need to define the api for js to do it. Also with include_paths. 

Issue:  #3745

File: tests/driver/nodejs/Cargo.toml (hunk 15 above)
Comment (FloVanGH) [reply tests/driver/nodejs/Cargo.toml]: This was a mistake and is fixed now.

File: api/node/package.json (hunk 12 above)
Comment (FloVanGH) [reply api/node/package.json:35]: I think for our use case we only need  `test`

Comment (FloVanGH) [reply api/node/package.json:35]: I removed that fields, because everything works without it. And we should add fields if we need it.

File: api/node/package.json (hunk 11 above)
Comment (FloVanGH) [reply api/node/package.json]: I removed the one we don't need.

File: api/node/package.json (hunk 9 above)
Comment (FloVanGH) [reply api/node/package.json]: @tronical do you mean with the `SEE LICNESE ... LICENSE.md`?

File: api/node/package.json (hunk 10 above)
Comment (FloVanGH) [reply api/node/package.json:10]: It looks like, that it works without it. @tronical should I remove it?

File: api/node/package.json (hunk 9 above)
Comment (tronical) [reply api/node/package.json]: Yes, this section "If you are using a license that hasn't been assigned an SPDX identifier, or if you are using a custom license, use a string value like this one".

File: api/node/package.json (hunk 10 above)
Comment (tronical) [reply api/node/package.json:10]: Yes. I'm in favor of keeping package.json minimal and only have fields that we need.

File: api/node/package.json (hunk 9 above)
Comment (FloVanGH) [reply api/node/package.json]: ok that's done.

File: api/node/package.json (hunk 10 above)
Comment (FloVanGH) [reply api/node/package.json]: Me ether ;-). Is removed.

File: api/node/package.json (hunk 8 above)
Comment (tronical) [reply api/node/package.json:6]: Re-opening this, as it is not done.

**Review by tronical — APPROVED** [review]
